atomic_scanners/misc-package-updates/scanner.py

```diff
@@ -0,0 +1,164 @@
+
+    if "DEPRECATION:" in outdated_packages[0]:
+        return outdated_packages[1:]
+
+    return outdated_packages
+
+
+def npm_list_of_outdated_packages(response):
+    # like pip, npm returns '\n' separated output but first line can be ignored
+    response = split_by_newline(response)[1:]
+
+    response = remove_last_blank_character(response)
+
+    return list_of_outdated_packages(response)
+
+
+def gem_list_of_outdated_packages(response):
+    # like pip, gem returns '\n' separated output
+    response = split_by_newline(response)
+
+    response = remove_last_blank_character(response)
+
+    return list_of_outdated_packages(response)
+
+
+def format_response(cli_arg, response):
+    if cli_arg == 'pip':
+        return pip_list_of_outdated_packages(response)
+    elif cli_arg == "npm":
+        return npm_list_of_outdated_packages(response)
+    elif cli_arg == "gem":
+        return gem_list_of_outdated_packages(response)
+
+
+def template_json_data(scan_type):
+    current_time = datetime.now().strftime('%Y-%m-%d-%H-%M-%S-%f')
+    json_out = {
+        "Start Time": current_time,
+        "Successful": "",
+        "Scan Type": scan_type + "-updates",
+        "UUID": UUID,
+        "CVE Feed Last Updated": "NA",
+        "Scanner": "Misc Package Updates",
+        "Scan Results": {"{} package updates".format(cli_arg): []}
+    }
+    return json_out
+
+json_out = template_json_data(cli_arg)
+
+try:
+    # Create the container before starting/running it
+    container = client.create_container(image=IMAGE_NAME,
+                                        command="tail -f /dev/null")
+
+    # Running the container
+    client.start(container.get('Id'))
+
+    # Check for pip updates
+    if cli_arg == "pip":
+        # variable to store info about exec_start
+        exe = client.exec_create(
+            container=container.get("Id"),
+            cmd="pip list --outdated"
+        )
+
+        response = client.exec_start(exe)
+
+    # Check for rubygem updates
+    elif cli_arg == "gem":
+        exe = client.exec_create(
+            container=container.get("Id"),
+            cmd="gem outdated"
+        )
+
+        response = client.exec_start(exe)
+
+    # Check for npm updates
+    elif cli_arg == "npm":
+        exe = client.exec_create(
+            container=container.get("Id"),
+            cmd="npm outdated"
+        )
+
+        response = client.exec_start(exe)
+
+    if binary_does_not_exist(response):
+        json_out["Scan Results"] = \
+            "Could not find {} executable in the image".format(cli_arg)
+    else:
+        json_out["Scan Results"]["{} package updates".format(cli_arg)] = \
+            format_response(cli_arg, response)
+        json_out["Finished Time"] = \
+            datetime.now().strftime('%Y-%m-%d-%H-%M-%S-%f')
+
+    # remove the container
+    client.remove_container(container=container.get("Id"), force=True)
+except Exception as e:
```

Comment:
log the exception /error message and we should put another error message about failure.